fixed simplified traffic, again

Please-Pick-a-Name opened the pull request:
- running without some body parts (bumpers, hood and trunk) now shows up as no body parts
- bx lost widebody (unlucky)
- etkc gone
- taxi fullsize and taxi lansdale (pre facelift only) works in simplified traffic
- legran now cares about facelift variance
- pickup now supports reg cab with normal bed, also now care about facelift variance
- roamer can be long now, and cares about facelift variance
- sunburst gone
- van now cares about facelift variance
- vivace gone
- wendover gone
- removed simplified traffic fallback logic (not gonna work now anyways)

- someone else can test this ig, i just wanna play some game at this point

Please-Pick-a-Name committed:
fixed simplified traffic, again

- why am i still here
- running without some body parts now shows up as no body parts
- bx lost widebody traffic
- etkc gone
- taxi fullsize and taxi lansdale (pre facelift only) works in simplified traffic
- legran now cares about facelift variance
- pickup now supports reg cab with normal bed, also now care about facelift variance
- roamer can be long now, and cares about facelift variance
- sunburst gone
- van now cares about facelift variance
- vivace gone
- wendover gone
- removed fallback logic (not gonna work anyways)

Please-Pick-a-Name commented:
sorry, i have commited a mistake
working on fixing it rn

Please-Pick-a-Name committed:
fixed simplified vehicle break vehicle config

Please-Pick-a-Name committed:
simplified vehicles now have wheels

- beam simplified vehicle have broken default wheels, so i implemented a wheel matcher without thinking about how i actually want it to be
- atleast it "works", you gets put in a standard steelie most of the time, but its wheel

Please-Pick-a-Name committed:
fix typo break van/pickup simple vehicle trim lvl

Please-Pick-a-Name commented:
simplified vehicle should work now again, i believe

O1LER reviewed (approved):
Seems to work as intended, tested on official servers, i guess running server v3.4.1, while using launcher v2.2.0

Was spectating random players and guests, noticed bastion and etk800 being replaced by their simple variants

WiserTixx commented:
Testing this PR, all the simplified variants seem to work except for the van and the roamer

Please-Pick-a-Name commented:
> Testing this PR, all the simplified variants seem to work except for the van and the roamer

quick question, how was it not working? the change vehicle failed or the facelift variance was mapped wrong?

WiserTixx commented:
> > Testing this PR, all the simplified variants seem to work except for the van and the roamer
> 
> quick question, how was it not working? the change vehicle failed or the facelift variance was mapped wrong?

My apologies, should have been more specific. The base van and roamer both don't spawn, resulting in a pink blob.

Please-Pick-a-Name commented:
> > > Testing this PR, all the simplified variants seem to work except for the van and the roamer
> > 
> > 
> > quick question, how was it not working? the change vehicle failed or the facelift variance was mapped wrong?
> 
> My apologies, should have been more specific. The base van and roamer both don't spawn, resulting in a pink blob.

I cannot replicate this on the roamer, unsure about the van cause i have yet to find anyone driving a van on beammp

Please-Pick-a-Name committed:
fixed nil slot handling

can happen on:
- pickup fascia, by removing rad support
- van fascia, by removing body
- van body, technically possible with modded frame, but not possible with vanilla

cannot happen on:
- paint_design
- pickup_frame, is main slot
- roamer frame, is main slot
for sanity sake, they get nil handling anyways

maybe fixes for roamer?

Please-Pick-a-Name commented:
attempted to fix pink blob under specific case, i think its feeding nil into string.find, could be something else tho

WiserTixx commented:
Both seem to work now.
